src/twisted/internet/defer.py

```diff
@@ -2129,7 +2129,11 @@ def handleCancel(result: Failure) -> Deferred[_T]:
 
         return status.deferred
 
-    _inlineCallbacks(None, gen, status, _copy_context())
+    # error: Argument 2 to "_inlineCallbacks" has incompatible type
+    # "Generator[Deferred[_T], object, _T] | Coroutine[Deferred[_T], object, _T]";
+    # expected
+    # "Generator[Deferred[_T], object, None] | Coroutine[Deferred[_T], object, None]"
+    _inlineCallbacks(None, gen, status, _copy_context())  # type: ignore[arg-type]
```

Comment:
@glyph any idea what the fix should be here?

`def _inlineCallbacks` was marked `@_extraneous` meaning it lost its arg-type making this call valid, fixing `@_extraneous` causes this arg-type error.

Reply:
YieldType (the first type arg) here is wrong. It should be `Deferred[object]`.  Saying that YieldType is `Deferred[_T]` and that matches ReturnType is saying that `async () -> int` using `Deferred`s can *only ever await upon `Deferred[int]`s* which is obviously silly.

I'm pretty sure we're wrong about this in a bunch of places and I probably propagated this error further in my recent type-hints upgrade.

Reply:
yep that fixed it for _inlineCallbacks